File: main/src/main/java/com/crossapplication/main/repository/interfaces/FoodCategoryRepository.java
package com.crossapplication.main.repository.interfaces;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.crossapplication.main.entity.FoodCategory;

@Repository
public interface FoodCategoryRepository extends JpaRepository<FoodCategory, Integer>{
    
    public List<FoodCategory> findByName(String name);

    public List<FoodCategory> findByNameContainingIgnoreCase(String keyword);
}

File: flutter/flutterfrontend/lib/domain/repositories/food_category_repository.dart
abstract class FoodCategoryRepository {
  Future<List<Map<String, dynamic>>> findByName(String name);
  Future<List<Map<String, dynamic>>> findByNameContainingIgnoreCase(String keyword);
  Future<Map<String, dynamic>?> findById(int id);
  Future<void> save(Map<String, dynamic> foodCategory);
  Future<void> update(int id, Map<String, dynamic> foodCategory);
  Future<void> delete(int id);
  Future<List<Map<String, dynamic>>> findAll();
}
